packages/as-types/assembly/amount.ts

```diff
@@ -28,62 +32,89 @@ export class Amount {
   /**
    * Adds two amounts and return results in a new one.
    *
-   * @param  a - Amount to add.
+   * @param  other - The other Amount to add.
+   *
+   * @returns
+   * - A Result containing the new Amount.
+   * - An error message if the currency is not the same between the two amounts.
+   * - An error message if the addition would overflow 'u64.MAX_VALUE'.
+   *
    */
-  add(a: Amount): Result<Amount> {
-    if (this.currency != a.currency) {
+  @operator('+')
+  add(other: Amount): Result<Amount> {
+    if (this.currency != other.currency) {
       return new Result(
         new Amount(),
         'the sum is impossible: the amounts have different currencies',
       );
     }
 
-    if (a.value > u64.MAX_VALUE - this.value) {
+    if (other.value > u64.MAX_VALUE - this.value) {
       // tests overflow
       return new Result(new Amount(), 'the sum is impossible: overflow');
     }
 
-    return new Result(new Amount(this.value + a.value, this.currency));
+    return new Result(new Amount(this.value + other.value, this.currency));
   }
 
   /**
-   * Substracts two amounts and return results in a new one.
+   * Subtracts two amounts and return results in a new one.
+   *
+   * @param  other - Amount to subtract.
    *
-   * @param  a - Amount to substract.
+   * @returns
+   * - A Result containing the new Amount.
+   * - An error message if the currency is not the same between the two amounts.
+   * - An error message if the subtraction would underflow .
```

Comment:
same as above, message is not clear, I have the felling that in some case I will get an error message where in fact I only get a Result